frontend/js/src/user/stats/components/UserArtistActivity.tsx

```diff
@@ -0,0 +1,147 @@
+import { ResponsiveBar } from "@nivo/bar";
+import * as React from "react";
+import { faExclamationCircle, faLink } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { useQuery } from "@tanstack/react-query";
+import Card from "../../../components/Card";
+import Loader from "../../../components/Loader";
+import { COLOR_BLACK } from "../../../utils/constants";
+import GlobalAppContext from "../../../utils/GlobalAppContext";
+
+export type UserArtistActivityProps = {
+  range: UserStatsAPIRange;
+  user: ListenBrainzUser;
+};
+
+export type UserArtistActivityState = {
+  data: UserArtistActivityResponse;
+  loading: boolean;
+  errorMessage: string;
+  hasError: boolean;
+};
+
+export default function UserArtistActivity({
+  range,
+  user,
+}: UserArtistActivityProps) {
+  const { APIService } = React.useContext(GlobalAppContext);
+
+  const { data: loaderData, isLoading: loading } = useQuery({
+    queryKey: ["userArtistActivity", user.name, range],
+    queryFn: async () => {
+      try {
+        const queryData = await APIService.getUserArtistActivity(
+          user.name,
+          range
+        );
+        return { data: queryData, hasError: false, errorMessage: "" };
+      } catch (error) {
+        return {
+          data: { result: [] } as UserArtistActivityResponse,
+          hasError: true,
+          errorMessage: error.message,
+        };
+      }
+    },
+  });
+
+  const {
+    data: rawData = { result: [] } as UserArtistActivityResponse,
+    hasError = false,
+    errorMessage = "",
+  } = loaderData || {};
+
+  const processData = (data?: UserArtistActivityResponse) => {
+    if (!data || !data.result || data.result.length === 0) {
+      return [];
+    }
+    return data.result.map((artist) => ({
+      label: artist.name.split(",")[0],
+      ...artist.albums.reduce(
+        (acc, album) => ({ ...acc, [album.name]: album.listen_count }),
+        {} as Record<string, number>
+      ),
+    })) as ChartDataItem[];
+  };
+
+  type ChartDataItem = {
+    label: string;
+  } & Record<string, number>;
+
```

Comment:
The type can be improved as:

```suggestion
  type ChartDataItem = {
    label: string;
    [albumName: string]: number | string;
  };
```